From: "Nikunj A. Dadhania" <nikunj.dadhania@intel.com>
Date: Mon, 27 Jan 2020 11:41:59 +0530
Subject: [PATCH] fix v5.4.15 merge

drivers/gpu/drm/i915/gt/intel_lrc.c
[ e22e3cf9aff0 drm/i915/execlists: Use per-process HWSP as scratch
Above commit removed slm_offset, adjust accordingly. ]

Signed-off-by: "Nikunj A. Dadhania" <nikunj.dadhania@intel.com>
---

--- a/drivers/gpu/drm/i915/gt/intel_lrc.c
+++ b/drivers/gpu/drm/i915/gt/intel_lrc.c
@@ -2711,10 +2711,10 @@
 	/* WaClearSlmSpaceAtContextSwitch:skl,bxt,kbl,glk,cfl */
 	batch = gen8_emit_pipe_control(batch,
 				       PIPE_CONTROL_FLUSH_L3 |
-				       PIPE_CONTROL_GLOBAL_GTT_IVB |
+				       PIPE_CONTROL_STORE_DATA_INDEX |
 				       PIPE_CONTROL_CS_STALL |
 				       PIPE_CONTROL_QW_WRITE,
-				       slm_offset(engine));
+				       LRC_PPHWSP_SCRATCH_ADDR);
 
 	batch = emit_lri(batch, lri, ARRAY_SIZE(lri));
 
